js/mexc3.js

```diff
@@ -4197,8 +4226,10 @@ module.exports = class mexc3 extends Exchange {
         let marginMode = undefined;
         [ marginMode, params ] = super.handleMarginModeAndParams (methodName, params);
         if (marginMode !== undefined) {
-            if (marginMode !== 'isolated') {
-                throw new NotSupported (this.id + ' only isolated margin is supported');
+            if (defaultType !== 'swap') {
```

Comment:
actually, I don't think we should be doing this, mexc loads all markets simultaneously (swap+spot), so the user might be trading derivatives with the defaultType set to `spot`

Currently this will fail but it shouldn't

```Javascript
exchange.options.defaultType = 'spot' // default value
exchange.createOrder("LTC/USDT:USDT", "market", "buy", 1, undefined, {"marginMode":"cross"})
```

I'd suggest adding another parameter to our `handleMarginModeAndParams` like `isSpot` that would be filled following these rules:
- if market available, isSpot=market['spot']
- otherwise isSpot = (defaultType === 'spot')

@Dan-krm wdyt?

Reply:
I think by moving the NotSupported error check to createSpotOrder we can avoid needing to add isSpot

Reply:
@Dan-krm sure, that will fix it too, thanks!